packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -123,7 +152,10 @@ class WeatherScreen extends HookWidget {
         ],
         body: SafeArea(
           child: RefreshIndicator(
-            onRefresh: load,
+            key: refreshIndicatorKey,
+            onRefresh: () async {
+              await fullWeatherStateNotifier.loadFullWeather();
+            },
```

Comment:
Please test this first:
```suggestion
            onRefresh: fullWeatherStateNotifier.loadFullWeather,
```